File: packages/jsii-pacmak/test/expected.java/java/org/jsii/tests/calculator/ObjectRefsInCollections.java
package org.jsii.tests.calculator;
/**
 * Verify that object references can be passed inside collections.
 */
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii$jsii_calc$.ObjectRefsInCollections")
public class ObjectRefsInCollections extends org.jsii.JsiiObject {
    protected ObjectRefsInCollections(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public ObjectRefsInCollections() {
        super(org.jsii.JsiiObject.InitializationMode.Jsii);
        org.jsii.JsiiEngine.getInstance().createNewObject(this);
    }
    /**
     * Returns the sum of all values
     */
    public java.lang.Number sumFromArray(final java.util.List<org.jsii.tests.calculator.lib.Value> values) {
        return this.jsiiCall("sumFromArray", java.lang.Number.class, java.util.stream.Stream.of(values).toArray());
    }
    /**
     * Returns the sum of all values in a map
     */
    public java.lang.Number sumFromMap(final java.util.Map<java.lang.String, org.jsii.tests.calculator.lib.Value> values) {
        return this.jsiiCall("sumFromMap", java.lang.Number.class, java.util.stream.Stream.of(values).toArray());
    }
}

File: packages/jsii-pacmak/test/expected.dotnet/AWS.Jsii.Tests.Calculator/ObjectRefsInCollections.cs
using AWS.Jsii.Runtime.Deputy;
using AWS.Jsii.Tests.Calculator.Lib;
using System.Collections.Generic;

namespace AWS.Jsii.Tests.Calculator
{
    /// <summary>Verify that object references can be passed inside collections.</summary>
    [JsiiClass(typeof(ObjectRefsInCollections), "jsii-calc.ObjectRefsInCollections", "[]")]
    public class ObjectRefsInCollections : DeputyBase
    {
        public ObjectRefsInCollections(): base(new DeputyProps(new object[]{}))
        {
        }

        protected ObjectRefsInCollections(ByRefValue reference): base(reference)
        {
        }

        protected ObjectRefsInCollections(DeputyProps props): base(props)
        {
        }

        /// <summary>Returns the sum of all values</summary>
        [JsiiMethod("sumFromArray", "{\"primitive\":\"number\"}", "[{\"name\":\"values\",\"type\":{\"collection\":{\"kind\":\"array\",\"elementtype\":{\"fqn\":\"@scope/jsii-calc-lib.Value\"}}}}]")]
        public virtual double SumFromArray(Value_[] values)
        {
            return InvokeMethod<double>(new object[]{values});
        }

        /// <summary>Returns the sum of all values in a map</summary>
        [JsiiMethod("sumFromMap", "{\"primitive\":\"number\"}", "[{\"name\":\"values\",\"type\":{\"collection\":{\"kind\":\"map\",\"elementtype\":{\"fqn\":\"@scope/jsii-calc-lib.Value\"}}}}]")]
        public virtual double SumFromMap(IDictionary<string, Value_> values)
        {
            return InvokeMethod<double>(new object[]{values});
        }
    }
}